vector/src/main/scala/geotrellis/vector/op/package.scala

```diff
@@ -1,7 +1,52 @@
 package geotrellis.vector
 
 package object op {
-  implicit class LineDissolveWrapper(val lines: Traversable[Line]) {
-    def dissolve() = LineDissolve.dissolve(lines)
+  /**
+   * An extension class - when in scope, these methods are available from Geometry objects.
+   *
+   * The algorithms herein are all implemented in JTS, but the wrapper methods
+   * here make it straightforward to call them with geotrellis.vector classes.
+   */
+  implicit class withGeometryMethods(val self: Geometry) extends MethodExtensions[Geometry] {
+    /**
+     * dissolve composes all the unique linestrings and points in the input
+     * geometry and produces a linear geometry with no repeated components.
+     */
+    def dissolve(): Geometry =
+      Geometry[Geometry](com.vividsolutions.jts.dissolve.LineDissolver.dissolve(self.jtsGeom))
+
+    /**
+     * convexHull finds the smallest convex polygon that contains all points of
+     * the input geometry
+     */
+    def convexHull(): Geometry =
+      Geometry[Geometry](self.jtsGeom.convexHull)
+
+    /**
+     * densify adds vertices to a Geometry without changing its shape.  The
+     * densified geometry may produce less distortion in operations such as
+     * reprojection.
+     *
+     * Point geometries and geometry components are unmodified by this operation.
+     *
+     * @param tolerance a distance in the same units as the coordinates of the
+     * Geometry.  This is a maximum distance between vertices for the result
+     * Geometry.
+     */
+    def densify(tolerance: Double): Geometry =
```

Comment:
It would be great to have this attached to the various types, so that it returned the logical type.
E.g. for a polygon, it would return a polygon and you wouldn't have to worry about checking the return type for the specific geometry type.